src/sonic-config-engine/minigraph.py

```diff
@@ -886,7 +921,26 @@ def parse_deviceinfo(meta, hwsku):
                 if desc != None:
                     port_descriptions[port_alias_map.get(alias, alias)] = desc.text
                 port_speeds[port_alias_map.get(alias, alias)] = speed
-    return port_speeds, port_descriptions
+
+            sysports = device_info.find(str(QName(ns, "SystemPorts")))
+            sys_ports = {}
+            if sysports is not None:
+                for sysport in sysports.findall(str(QName(ns, "SystemPort"))):
+                    slot_id = sysport.find(str(QName(ns, "SlotId")))
+                    portname = sysport.find(str(QName(ns, "Name"))).text
+                    device = sysport.find(str(QName(ns, "Device"))).text
+                    system_port_id = sysport.find(str(QName(ns, "SystemPortId"))).text
+                    switch_id = sysport.find(str(QName(ns, "SwitchId"))).text
+                    core_id = sysport.find(str(QName(ns, "CoreId"))).text
+                    core_port_id = sysport.find(str(QName(ns, "CorePortId"))).text
+                    speed = sysport.find(str(QName(ns, "Speed"))).text
+                    num_voq = sysport.find(str(QName(ns, "NumVoq"))).text
+                    key = "%s|%s" % (device, portname)
+                    if slot_id is not None:
```

Comment:
Isn't slot_id a required attribute ? will handling of system ports in swss work, if slot-id is not there in the key ?

Reply:
`slot_id` is required to identify each linecard in a chassis. For a fixed system with multiple asics this is not required since the `device` is enough to identify the each asic of the system.

In the end `slot_id` is only necessary to build a unique key to identify each asic in the system. If that helps we can make it mandatory for single-asic systems even if this is not required.